shell.go

```diff
@@ -7,6 +7,11 @@ import (
 	"unsafe"
 )
 
+// Windows' constants.
+const (
+	windowsSideError uint32 = 4294967295 // Underflowed -1
```

Comment:
Can we do a binary operation with the max size of uint32?

Reply:
math.MaxUint32 looks simpler

Reply:
ah, and you do remove it with the "always return an error" case, I’m eager to see if you still differentiate WSL/Windows or internal side.

Reply:
The error messages are different, and end users can use `(*ShellError).ExitCode` to check.

Honestly I'm less and less sure we should be explicit about it, because this is not something documented but rather observed behaviour.

Reply:
as long as we return any error if something fails, it’s fine, I don’t think that the user will then go deeper (user being the api user ofc)